docs/hub/datasets-pyarrow.md

```diff
@@ -0,0 +1,234 @@
+
+```python
+>>> import pyarrow.parquet as pq
+>>> table = pq.read_table("hf://datasets/stanfordnlp/imdb/plain_text/train-00000-of-00001.parquet")
+>>> table
+pyarrow.Table
+text: string
+label: int64
+----
+text: [["I rented I AM CURIOUS-YELLOW from my video store because of all the controversy that surrounded it (... 1542 chars omitted)", ...],...,[..., "The story centers around Barry McKenzie who must go to England if he wishes to claim his inheritan (... 221 chars omitted)"]]
+label: [[0,0,0,0,0,...,0,0,0,0,0],...,[1,1,1,1,1,...,1,1,1,1,1]]
+```
+
+If you don't want to load the full Parquet data, you can get the Parquet metadata or load row group by row group instead:
+
+```python
+>>> import pyarrow.parquet as pq
+>>> pf = pq.ParquetFile("hf://datasets/stanfordnlp/imdb/plain_text/train-00000-of-00001.parquet")
+>>> pf.metadata
+<pyarrow._parquet.FileMetaData object at 0x1171b4090>
+  created_by: parquet-cpp-arrow version 12.0.0
+  num_columns: 2
+  num_rows: 25000
+  num_row_groups: 25
+  format_version: 2.6
+  serialized_size: 62036
+>>> for i in pf.num_row_groups:
+...     table = pf.read_row_group(i)
+...     ...
+```
+
+For more information on the Hugging Face paths and how they are implemented, please refer to the [the client library's documentation on the HfFileSystem](/docs/huggingface_hub/guides/hf_file_system).
+
+## Save a Table
+
+You can save a pyarrow Table using `pyarrow.parquet.write_table` to a local file or to Hugging Face directly.
+
+To save the Table on Hugging Face, you first need to [Login with your Hugging Face account](/docs/huggingface_hub/quick-start#login), for example using:
+
+```
+huggingface-cli login
+```
+
+Then you can [Create a dataset repository](/docs/huggingface_hub/quick-start#create-a-repository), for example using:
+
+```python
+from huggingface_hub import HfApi
+
+HfApi().create_repo(repo_id="username/my_dataset", repo_type="dataset")
+```
+
+Finally, you can use [Hugging Face paths](/docs/huggingface_hub/guides/hf_file_system#integrations) in PyArrow:
+
+```python
+import pyarrow.parquet as pq
+
+pq.write_table(table, "hf://datasets/username/my_dataset/imdb.parquet", use_content_defined_chunking=True)
+
+# or write in separate files if the dataset has train/validation/test splits
+pq.write_table(table_train, "hf://datasets/username/my_dataset/train.parquet", use_content_defined_chunking=True)
+pq.write_table(table_valid, "hf://datasets/username/my_dataset/validation.parquet", use_content_defined_chunking=True)
+pq.write_table(table_test , "hf://datasets/username/my_dataset/test.parquet", use_content_defined_chunking=True)
+```
+
+We use `use_content_defined_chunking=True` to enable faster uploads and downloads from Hugging Face thanks to Xet deduplication (it requires `pyarrow>=21.0`).
```

Comment:
do you think we could ask them to turn it on by default?

Reply:
hard to say, wdyt @kszucs ?

Reply:
Maybe in case of a `hf://` URI but that could be fragile to implement since filesystem objects can be passed as well. 
I would rather consider the `flavor` argument https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html

```py
pq.write_table(table_test , "hf://datasets/username/my_dataset/test.parquet", flavor="huggingface")
```

Reply:
Also note that CDC has a slight performance overhead, so turning it on by default would make sense if we can infer that the targeted filesystem can use it for deduplication.